src/main/java/seedu/elisa/model/ItemModelManager.java

```diff
@@ -363,18 +358,62 @@ public void replaceItem(Item item, Item newItem) {
             }
         }
 
+        System.out.println("Boo");
+        System.out.println(reminderList.indexOf(item));
         if ((index = reminderList.indexOf(item)) >= 0) {
             if (newItem.hasReminder()) {
                 reminderList.setItem(index, newItem);
             } else {
                 reminderList.remove(index);
             }
 
-            //Old reminder must be in active since it already rang.
-            //Find the old reminder from activeReminders and remove it.
-            activeReminders.remove(item);
-            //Put the new reminder in futureReminders if it will be rung later.
-            futureReminders.add(newItem);
+            System.out.println("Help");
+
+            if (!item.getReminder().equals(newItem.getReminder())) {
+                System.out.println("Hello?");
+                if (newItem.getReminder().isEmpty()) {
+                    System.out.println("Shabam");
+                    if (activeReminders.contains(item)) {
+                        activeReminders.remove(item);
+                    }
+
+                    if (futureReminders.contains(item)) {
+                        futureReminders.remove(item);
+                    }
+
+                } else {
+                    //means that newItem has a reminder that is diff from old reminder
+                    //if old had nothing just add to corresponding
+                    //if old had a reminder, remove it
+                    if (item.getReminder().isPresent()) {
+                        System.out.println("Hi");
+                        if (activeReminders.contains(item)) {
+                            activeReminders.remove(item);
+                        }
+
+                        if (futureReminders.contains(item)) {
+                            futureReminders.remove(item);
+                            System.out.println("Removed");
+                        }
+                    }
+                    Reminder newReminder = newItem.getReminder().get();
+
+                    if (newReminder.getOccurrenceDateTime().isAfter(LocalDateTime.now())) {
+                        //Add to futureReminders if it is to occur later.
+                        futureReminders.add(newItem);
+                        //Otherwise do not add
+                    }
+                }
+            }
+        }
+
+        System.out.println(index);
+        if (index >= 0) {
+            if (visualList.belongToList(newItem)) {
+                visualList.setItem(index, newItem);
+            } else {
+                visualList.remove(index);
+            }
```

Comment:
You cant just copy and paste this block of code actually, cause if you look at the individual loop, the index of the item is actually getting updated at each round. So before this block you need: `index = visualList.indexOf(item)`

Reply:
Nice catch! I edited this as well as the initialization of the index to match the first check.